packages/ui-i18n/src/bidirectional.tsx

```diff
@@ -21,12 +21,28 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import React, { forwardRef } from 'react'
+import React, { ComponentClass, ForwardedRef, forwardRef } from 'react'
 import { decorator } from '@instructure/ui-decorator'
 import { DIRECTION, TextDirectionContext } from './TextDirectionContext'
-// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'hois... Remove this comment to see the full error message
 import hoistNonReactStatics from 'hoist-non-react-statics'
 
+// This is a workaround because TS cannot take type information from
+// decorators into account. This type needs to be added to every component,
+// that uses the bidirectional decorator.
+// see https://github.com/microsoft/TypeScript/issues/4881
+export type BidirectionalProps = {
+  dir?: 'ltr' | 'rtl'
+}
```

Comment:
How do you like this pattern? Its a workaround, but I could not think of anything better

Reply:
I did something similar with the the `withStyle` decorator in my PR [here](https://github.com/instructure/instructure-ui/blob/8c83dcc9c1174ad469e4f5e90042572754196b26/packages/emotion/src/withStyle.tsx#L37-L40,  so I think this is a reasonable solution, but in the long run I think it would be the best to rewrite all of our current decorators to [Higher Order Components](https://reactjs.org/docs/higher-order-components.html) and hooks for the functional components.